californium-core/src/main/java/org/eclipse/californium/core/coap/BlockOption.java

```diff
@@ -235,6 +235,14 @@ public boolean equals(Object o) {
 		return szx == block.szx && num == block.num && m == block.m;
 	}
 
+	@Override
+	public int hashCode() {
+		int result = szx;
+		result = 31 * result + (m ? 1 : 0);
+		result = 31 * result + num;
+		return result;
+	}
```

Comment:
I suppose the idea is to say that blockOption is now immutable ? 
- Should we still need setter (as there are private now)?
- Should we not define szx, num as final ?